iree/compiler/InputConversion/MHLO/Passes.cpp

```diff
@@ -64,6 +64,8 @@ void buildMHLOInputConversionPassPipeline(OpPassManager &passManager) {
       mlir::iree_compiler::createConvertMHLOToLinalgExtPass());
   passManager.addNestedPass<FuncOp>(
       mlir::iree_compiler::createMHLOToLinalgOnTensorsPass());
+  passManager.addNestedPass<FuncOp>(
+      mlir::iree_compiler::createPromoteI1ToI8Pass());
```

Comment:
What prompted this change? I recently removed a similar pass from this file in https://github.com/google/iree/pull/6852 based on discussion here: https://github.com/google/iree/issues/6756#issuecomment-898607883. We really need a clear stance (that is documented) on how i1 types should be handled (https://github.com/google/iree/issues/5223).

Reply:
There are i1 constants (e.g., tensor<1024xi1>), and IREE does not know how to handle i1 constants.

I think it is slightly different from #6852 because PromoteTensorLoads is mainly to handle 0D i1 tensors (i.e. tensor<i1>) which are introduced by CFG.

I would see this change in these ways:

1. The pass is already used in TOSA pipeline. This makes the MHLO pipeline align the TOSA pipeline. I.e., the pass is added because IREE does not handle i1 constants correctly now.
2. The i1 constants could be used in computations. Ideally, IREE should pack constants, and codegen side should be able to handle it. We will see similar situations in quantized model, e.g., `tensor<1024xi4>`, though the quant stories are not clear to me at this moment.

To be more clear, I think this case should be handled like:

1. At flow level, the host side decides how to pack/store these non-native types (e.g., `i1`, `i4`).
2. IREE compiler will configure codegen side to store constants in some types. E.g., today we are storing i1 in 8 bits on SPIR-V side. See https://github.com/llvm/llvm-project/blob/a95ba8107359e17cb1669c01f416fd2723a23126/mlir/include/mlir/Dialect/SPIRV/Transforms/SPIRVConversion.h#L55-L63

Once IREE HAL can store these types correctly (maybe pack them back-to-back), and all the backends can emulate the case. I think we will have a clearer picture.